rust/blockstore/src/arrow/provider.rs

```diff
@@ -252,14 +254,14 @@ impl BlockManager {
         Block::from_record_batch(delta_id, record_batch)
     }
 
-    pub(super) fn cached(&self, id: &Uuid) -> bool {
-        self.block_cache.get(id).is_some()
+    pub(super) async fn cached(&self, id: &Uuid) -> bool {
+        self.block_cache.get(id).await.ok().is_some()
     }
 
     pub(super) async fn get(&self, id: &Uuid) -> Result<Option<Block>, GetError> {
-        let block = self.block_cache.get(id);
+        let block = self.block_cache.get(id).await.ok().flatten();
```

Comment:
This means every access to the cache results in a copying of the data in the block, which means basically all queries copy potentially 100's of MB of data around.

Reply:
+1, I agree that it seems like every cache access converts bytes to blocks which copies data (a block is 8MB so 8MB per access) and vice versa (unless I am missing something which is also likely)